spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1789,15 +1826,15 @@ private void restoreProperties(Message message, PendingReply pendingReply) {
 		}
 	}
 
-	private QueueingConsumer createQueueingConsumer(final String queueName, Channel channel) throws Exception {
+	private DefaultConsumer createConsumer(final String queueName, Channel channel,
+			CompletableFuture<Delivery> future) throws Exception {
 		channel.basicQos(1);
 		final CountDownLatch latch = new CountDownLatch(1);
-		QueueingConsumer consumer = new QueueingConsumer(channel) {
+		DefaultConsumer consumer = new DefaultConsumer(channel) {
 
 			@Override
 			public void handleCancel(String consumerTag) throws IOException {
-				super.handleCancel(consumerTag);
-				latch.countDown();
+				future.completeExceptionally(new ConsumerCancelledException());
```

Comment:
Hm... I think we should introduce some context to the exception to determine which consumer was canceled.
At least mention that `consumerTag`.